### PR TITLE
tests(navsets): Add navsets kitchensink tests

### PR DESCRIPTION
**Pull request by karangattu**
Also, address other open TODO items along the way

TODO:
- [x] Test placement parameter
- [x] Add app for navset_bar
- [x] Add test for navset_bar
- [x] Add control for navset_bar
- [x] Fixes https://github.com/posit-dev/py-shiny/issues/1609

### CODE REVIEW

File: shiny/playwright/controller/_controls.py [hunk 1]
```diff
@@ -1587,7 +1587,6 @@ def set(self, value: bool, *, timeout: Timeout = None, **kwargs: object) -> None
             value, timeout=timeout, **kwargs  # pyright: ignore[reportArgumentType]
         )
 
-    # TODO-karan-test: Convert usage of _toggle() to set()
     def _toggle(self, *, timeout: Timeout = None, **kwargs: object) -> None:
```
Comment (schloerke) [inline shiny/playwright/controller/_controls.py:1590]: Can this method be removed?

File: shiny/playwright/controller/_controls.py [hunk 2]
```diff
@@ -5500,6 +5508,21 @@ def get_loc_active_content(self, *, timeout: Timeout = None) -> Locator:
             f"div.tab-content[data-tabsetid='{datatab_id}'] > div.tab-pane.active"
         )
 
+    def expect_title(self, value: PatternOrStr, *, timeout: Timeout = None) -> None:
```
Comment (schloerke) [inline shiny/playwright/controller/_controls.py:5511]: We should move this to a mixin as not every navset contains `title=`

File: shiny/playwright/controller/_controls.py [hunk 3]
```diff
@@ -5577,7 +5600,7 @@ class NavPanel(_UiWithContainer):
     Playwright `Locator` for the nav panel container.
     """
```
Comment (schloerke) [inline shiny/playwright/controller/_controls.py:5602]:
```suggestion

    panel_value
    """
    The `data-value` attribute used to identify the nav panel within the larger navset.
    """
```

File: shiny/playwright/controller/_controls.py [hunk 4]
```diff
@@ -5820,8 +5861,45 @@ def __init__(self, page: Page, id: str) -> None:
             loc="> li.nav-item",
         )
 
+    def expect_placement(
```
Comment (schloerke) [inline shiny/playwright/controller/_controls.py:5864]: Let's move this to another mixin (`NavsetPlacementM`) and add it to both `NavsetCardUnderline` and `NavsetCardPill`

File: shiny/playwright/controller/_controls.py [hunk 5]
```diff
@@ -5820,8 +5861,45 @@ def __init__(self, page: Page, id: str) -> None:
...
+        Parameters
+        ----------
+        sidebar
+            `True` if the sidebar is expected to be visible, `False` otherwise.
+        timeout
+            The maximum time to wait for the expectation to pass. Defaults to `None`.
+        """
+        playwright_expect(
+            self.loc_container.locator("..").locator("+ .bslib-sidebar-layout")
+        ).to_have_count(int(sidebar), timeout=timeout)
```
Comment (schloerke) [inline shiny/playwright/controller/_controls.py:5899]:
```suggestion
    def expect_sidebar(self, has_sidebar: bool, *, timeout: Timeout = None) -> None:
        """
        Assert whether or not the sidebar exists within the navset.

        Parameters
        ----------
        sidebar
            `True` if the sidebar exists within the navset.
        timeout
            The maximum time to wait for the expectation to pass. Defaults to `None`.
        """
        playwright_expect(
            self.loc_container.locator("..").locator("+ .bslib-sidebar-layout")
        ).to_have_count(int(sidebar), timeout=timeout)
```

File: shiny/playwright/controller/_controls.py [hunk 6]
```diff
@@ -5820,8 +5861,45 @@ def __init__(self, page: Page, id: str) -> None:
...
+        if location == "below":
+            playwright_expect(self.loc_container.locator("..")).to_have_class(
+                "card-footer", timeout=timeout
+            )
+        else:
+            playwright_expect(self.loc_container.locator("..")).to_have_class(
+                "card-header", timeout=timeout
+            )
+
+    def expect_sidebar(self, sidebar: bool, *, timeout: Timeout = None) -> None:
```
Comment (schloerke) [inline shiny/playwright/controller/_controls.py:5886]: Please add this to `NavsetCardPill` and `NavsetCardTab`. If a duplicate in code, please use a mixing class (`NavsetCardSidebarM`?)

Comment (schloerke) [inline shiny/playwright/controller/_controls.py:5886]: Please add a `.loc_sidebar` property to all classes that contain `.expect_sidebar()`

File: shiny/playwright/controller/_controls.py [hunk 7]
```diff
@@ -5864,6 +5942,124 @@ def __init__(self, page: Page, id: str) -> None:
...
+        Parameters
+        ----------
+        sidebar
+            `True` if the sidebar is expected to be visible, `False` otherwise.
+        timeout
+            The maximum time to wait for the expectation to pass. Defaults to `None`.
+        """
+        if sidebar:
+            playwright_expect(
+                self.get_loc_active_content().locator("..").locator("..").locator("..")
```
Comment (schloerke) [inline shiny/playwright/controller/_controls.py:6019]: We should pair to make a static property for the `.loc_sidebar`

File: shiny/playwright/controller/_controls.py (hunk 4 above)
Comment (karangattu) [reply shiny/playwright/controller/_controls.py:5942]: I see `placement` for [NavsetCardUndeline](https://shiny.posit.co/py/api/express/express.ui.navset_card_underline.html) but not for [NavsetCardPill](https://shiny.posit.co/py/api/express/express.ui.navset_card_pill.html)
Let me know if I am not looking in the right place

Comment (schloerke) [reply shiny/playwright/controller/_controls.py:5942]: It should be there. Added it in this PR. 

I'll finish out the mixin

**Pull request merged**